joomla/joomla-cms
# [#17744] - Pressing clear button of calendar on required date field submits admin form in Joomla! 3.7.5 --- corrected

Pull Request for Issue # 17744

### Summary of Changes
Previously, when we edit an article under publishing tab, the clear button in calendar field did not work if it was a required field.
Now, it clears the calendar field when pressed instead of redirecting (as it did originally).
Changes were made in calendar.js and calendar.php

### Testing Instructions
Create an admin form with a required date field. i.e. open administrator\components\com_content\models\forms\article.xml and make field "publish_down" required by adding required="true" to field's xml node.
Go to Joomla! administration panel and open an article edit view.
Finish Publishing field should now be a required field.
Add Finish Publishing date, using the date picker of the calendar
Click "Clear" button on the Calendar.
